Makefile

```diff
@@ -57,23 +57,39 @@ CXXFLAGS += -Wall -Wextra -ggdb -I. -I"$(YOSYS_SRC)" -MD -D_YOSYS_ -fPIC -I$(PRE
 LDFLAGS += -L$(LIBDIR)
 LDLIBS = -lstdc++ -lm
 
-PKG_CONFIG = pkg-config
-SED = sed
-BISON = bison
+PKG_CONFIG ?= pkg-config
+SED ?= sed
+BISON ?= bison
 
 ifeq (Darwin,$(findstring Darwin,$(shell uname)))
-	# add macports/homebrew include and library path to search directories, don't use '-rdynamic' and '-lrt':
-	CXXFLAGS += -I/opt/local/include -I/usr/local/opt/readline/include
-	LDFLAGS += -L/opt/local/lib -L/usr/local/opt/readline/lib
-	# add homebrew's libffi include and library path
-	CXXFLAGS += $(shell PKG_CONFIG_PATH=$$(brew list libffi | grep pkgconfig | xargs dirname) pkg-config --silence-errors --cflags libffi)
-	LDFLAGS += $(shell PKG_CONFIG_PATH=$$(brew list libffi | grep pkgconfig | xargs dirname) pkg-config --silence-errors --libs libffi)
-	# use bison installed by homebrew if available
-	BISON = $(shell (brew list bison | grep -m1 "bin/bison") || echo bison)
-	SED = sed
+
+# homebrew search paths
+ifneq ($(shell which brew),)
+BREW_PREFIX := $(shell brew --prefix)/opt
+
+CXXFLAGS += -I$(BREW_PREFIX)/readline/include
+LDFLAGS += -L$(BREW_PREFIX)/readline/lib
+
+PKG_CONFIG_PATH := $(BREW_PREFIX)/libffi/lib/pkgconfig:$(PKG_CONFIG_PATH)
+PKG_CONFIG_PATH := $(BREW_PREFIX)/tcl-tk/lib/pkgconfig:$(PKG_CONFIG_PATH)
```

Comment:
I get a build error when building after I `brew cask install tcl`.
It builds fine without homebrew tcl.

```
...
clang++ -o yosys-filterlib -L/usr/local/lib -L/Users/jheadapohl/Documents/homebrew/opt/readline/lib passes/techmap/filterlib.o -lstdc++ -lm -lreadline -L/Users/jheadapohl/Documents/homebrew/Cellar/libffi/3.0.13/lib -lffi -ldl -ltcl8.6
cp techlibs/ice40/brams_init1.vh share/ice40/brams_init1.vh
cp techlibs/ice40/brams_init2.vh share/ice40/brams_init2.vh
mkdir -p share/ice40
mkdir -p share/xilinx
cp techlibs/ice40/brams_init3.vh share/ice40/brams_init3.vh
cp techlibs/xilinx/brams_init_36.vh share/xilinx/brams_init_36.vh
mkdir -p share/xilinx
mkdir -p share/xilinx
ld: library not found for -ltcl8.6
clang: error: linker command failed with exit code 1 (use -v to see invocation)
cp techlibs/xilinx/brams_init_32.vh share/xilinx/brams_init_32.vh
make: *** [yosys-filterlib] Error 1
make: *** Waiting for unfinished jobs....
cp techlibs/xilinx/brams_init_18.vh share/xilinx/brams_init_18.vh
```